library/src/main/java/com/novoda/downloadmanager/InternalFilePersistence.java

```diff
@@ -35,23 +36,30 @@ public FilePersistenceResult create(FileName fileName, FileSize fileSize) {
             return FilePersistenceResult.newInstance(Status.ERROR_INSUFFICIENT_SPACE);
         }
 
-        FilePath filePath = FilePathCreator.create(fileName.name());
+        File internalFileDir = context.getFilesDir();
+
+        String absolutePath = internalFileDir.getAbsolutePath() + File.separatorChar + fileName.name();
```

Comment:
What about `new File(internalFileDir, fileName.name()).getAbsolutePath()`?

I think we should not create paths by concatenating strings

Reply:
Good point, done in 3074d45